### DISCUSSION TITLE
Does ytt distinguish between floats and integers?

### DISCUSSION THREAD

**Discussion by maxwell-k** [k14s/ytt#844] [Q&A] [answered]
While evaluating ytt as a potential component in configuration management for maritime vessel simulation I noticed that it will convert floating point numbers to integers. I initially thought that might be [an issue] but that was closed.

Now I'm wondering is that a deliberate design decision?  How does ytt handle the different [types in yaml]?

[types in yaml]: https://yaml.org/spec/1.2.2/#102-json-schema
[an issue]: https://github.com/carvel-dev/ytt/issues/821

**Comment by 100mik** [comment, marked as answer]
Thanks for bumping this! From the looks of it the issue was missed and closed by GitHub automation. This was _not_ intentional.
Let's continue the discussion on the issue itself. Apologies from our end 🙏🏼 

I will leave this discussion open for now, and it would be super cool if you could add some details about how you are trying to use `ytt`. Looks like you have a really niche but fun use case!

**Reply by maxwell-k** [reply]
Thank you for the helpful response!  A few details of how we are trying to use ytt:

The rough context is that we are considering ytt as a potential component in a system for modelling maritime vessels. The relevant design of the vessel is stored in YAML.  We have a workflow where we change a design parameter and then simulate a journey made by the vessel. 

The system currently errors if the processed YAML includes an integer as input where a float is expected. I appreciate that in some other use cases these may be interchangeable. For the current version of this application they are not. The unexpected conversion makes it difficult or impossible to use ytt as a part of the system.
